stackdriver/client_test.go

```diff
@@ -114,3 +117,43 @@ func TestEmptyRequest(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestResolver(t *testing.T) {
+  grpcServer := grpc.NewServer()
+	listener := newLocalListener()
+	go grpcServer.Serve(listener)
+	defer grpcServer.Stop()
+
+	serverURL, err := url.Parse("http://stackdriver.invalid?auth=false")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	res, _ := manual.GenerateAndRegisterManualResolver()
+	res.InitialAddrs([]resolver.Address{
+		{Addr: "localhost"},
+	})
+	c := NewClient(&ClientConfig{
+		URL:      serverURL,
+		Timeout:  time.Second,
+		Resolver: res,
+	})
+
+  address := c.url.Hostname()
+	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+	defer cancel()
+
+	c.resolver.Scheme()
+
+	conn, connerr := grpc.DialContext(ctx, address, grpc.WithInsecure())
+	c.conn = conn
+	if connerr != nil {
+		t.Fatal(connerr)
+	}
```

Comment:
This test will pass in some cases when `DialContext` fails, because by default this method is non-blocking. Try passing the grpc.WithBlock() option. I'm guessing this is the reason the test passes even though the method has a bug in how it uses the resolver (see my comment in client.go).

Depending on the transport implementation this may not do anything until you call `Invoke` on `conn`, but I think you'll find out as soon as you start calling this with `WithBlock`.